main.go

```diff
@@ -75,10 +76,14 @@ func sourceFromArg(arg string) (*source, error) {
 	}
 
 	// a GitHub or GitLab URL (even without the protocol):
+	var dnsErr *net.DNSError
 	src, err := readmeURL(arg)
 	if src != nil && err == nil {
 		// if there's an error, try next methods...
 		return src, nil
+	} else if err != nil && errors.As(err, &dnsErr) {
+		// fail if there's a network error.
+		return src, err
```

Comment:
The code could be clearer

```suggestion
	// a GitHub or GitLab URL (even without the protocol):
	src, err := readmeURL(arg)
	if src != nil && err == nil {
		// if there's an error, try next methods...
		return src, nil
	}
	
	var dnsErr *net.DNSError
	if errors.As(err, &dnsErr) {
		// fail if there's a network error.
		return src, err
```

- The `else` is not needed as you use `return` in the `if`
- no need to check `err != nil && errors.As(err`, see https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/errors/wrap.go;l=97-100